### ISSUE TITLE
dove + knex unable to use $like, $ilike and $notlike

### ISSUE THREAD

**Issue by alwex** [feathersjs/feathers#2759]
### Steps to reproduce

![image](https://user-images.githubusercontent.com/63174/191658005-77aaa0a5-e82c-427a-a04e-0ed60bab6ded.png)

when using a generated service and trying to use knex operators `$like`, `$ilike` and `$notlike` typescript completion does not complete and the validation fail for the query

```
error: validation failed {"className":"bad-request","code":400,"data":[{"instancePath":"/email","keyword":"type","message":"must be string","params":{"type":"string"},"schemaPath":"#/properties/email/anyOf/0/type"},{"instancePath":"/email","keyword":"additionalProperties","message":"must NOT have additional properties","params":{"additionalProperty":"$and"},"schemaPath":"#/properties/email/anyOf/1/additionalProperties"},{"instancePath":"/email","keyword":"anyOf","message":"must match a schema in anyOf","params":{},"schemaPath":"#/properties/email/anyOf"}],"name":"BadRequest","type":"FeathersError"}
```
### Expected behavior
Knex operators are working fine and the api returns the correct result

### Actual behavior
validation fail and completion does not work properly

I use feathersjs `5.0.0-pre.29`

![image](https://user-images.githubusercontent.com/63174/191658250-139bd773-dc32-48dd-8b34-aee4ee9bc1cb.png)

**Comment by tvld** [comment]
I read something  about [querySyntax](https://dove.feathersjs.com/api/schema/typebox.html#querysyntax) and ``$ilike`` ?

**Comment by daffl** [comment]
There is an example how to allow this in https://dove.feathersjs.com/api/schema/typebox.html#querysyntax

**Comment by daffl** [comment]
#2931 will also have a better way to add additional properties to the query syntax:

```ts
    // This will additionally allow querying for `{ message: { $ilike: 'He%' } }`
    querySyntax(messageQueryProperties, {
      text: {
        $ilike: Type.String()
      }
    })
```

**Comment by tvld** [comment]
@daffl  is there some kind of regression in v34? 

This:
```ts
querySyntax(messageQueryProperties, {
      text: {
        $ilike: Type.String()
      }
    })
```

Gives Error : 
```
(alias) const Type: TypeBuilder
import Type
JSON Schema Type Builder with Static Type Resolution for TypeScript

Expected 1 arguments, but got 2  
```

**Comment by daffl** [comment]
This has been published now in `5.0.0-pre.35`
